echopop/survey.py

```diff
@@ -1264,98 +1272,113 @@ def apportion_kriged_biomass( self ,
-                                                                  length_df_copy ,
-                                                                  weight_strata ,
-                                                                  weight_strata_station ,
-                                                                  aged_proportions ,
-                                                                  length_intervals )
-
-        ### Parse kriged biomass (areal density) values
-        kriged_results = self.statistics[ 'kriging' ][ 'kriged_biomass_df' ]
-
-        kriged_results = (
-            kriged_results
-            .merge( dist_weight_sum , on = [ 'stratum_num' ] )
-            .assign( biomass_aged = lambda df: df.B_a_adult_mean * df.proportion_normalized * df.cell_area_nmi2 )
-            .merge( haul_sex_weights_normalized , on = [ 'stratum_num' , 'sex' ] )
-            .assign( biomass_unaged = lambda df: df.B_a_adult_mean * df.proportion_normalized_station_1 * df.cell_area_nmi2 ,
-                     biomass_total = lambda df: df.biomass_aged + df.biomass_unaged )
-        )
+        
+        # ---- Construct the complete dataframe containing all possible
+        # ---- strata, sexes, lengths, and ages
+        # full_biological_indices = pd.DataFrame(
+        #     list( product( np.unique( specimen_spp.stratum_num ) ,
+        #                    np.unique( specimen_spp.species_id ) ,
+        #                    np.unique( specimen_spp.sex ) ,
+        #                    pd.cut( length_bins , length_intervals ) ,
+        #                    pd.cut( age_bins , age_intervals ) ) ) ,
+        #     columns = [ 'stratum_num' , 'species_id' , 'sex' , 
+        #                 'length_bin' , 'age_bin' ]
+        # )
 
-        ### Parse the average weight calculations
-        average_weight_df = (
-            self.biology[ 'weight' ][ 'weight_strata_df' ]
-            .loc[ : , [ 'stratum_num' , 'average_weight_total' ] ]
-        )
+        ### Process `haul_spp`
+        # Remove haul numbers not found within `length_spp`
+        haul_spp_matched = haul_spp[ haul_spp.haul_num.isin( length_spp.haul_num ) ]
 
-        ### Merge with the kriged results 
-        kriged_results = (
-            kriged_results
-            .merge( average_weight_df , on = [ 'stratum_num' ] )
-            .assign( N_adult_kriged = lambda df: df.B_adult_kriged / df.average_weight_total )
-            .merge( self.acoustics[ 'sigma_bs' ][ 'strata_mean' ] , on = [ 'stratum_num' ] )
-            .assign( NASC_kriged = lambda df: df.N_adult_kriged * df.sigma_bs_mean )
+        ### Sum weights for aged/unaged and all within each stratum
+        weight_strata_aged_unaged , weight_strata = sum_strata_weight( haul_spp_matched ,
+                                                                       specimen_spp )
+        
+        ### Calculate the summed aged proportions for all aged fish, and for aged fish        
```

Comment:
do you mean for aged and UNaged fish?
```suggestion
        ### Calculate the summed aged proportions for all aged fish and unaged fish        
```